src/main/java/roomescape/payment/infrastructure/TossPaymentClient.java

```diff
@@ -0,0 +1,52 @@
+package roomescape.payment.infrastructure;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClient;
+import roomescape.payment.application.PaymentClient;
+import roomescape.payment.application.dto.PaymentRequest;
+import roomescape.payment.application.dto.PaymentResponse;
+import roomescape.payment.infrastructure.dto.TossErrorResponse;
+import roomescape.payment.infrastructure.dto.TossPaymentRequest;
+
+@RequiredArgsConstructor
+@Component
+public class TossPaymentClient implements PaymentClient {
+
+    private final RestClient restClient;
+
+    @Value("${payment.toss.secret-key}")
+    private String secretKey;
+
+    public PaymentResponse requestPayment(final PaymentRequest request) {
+        String encodedKey = Base64.getEncoder().encodeToString((secretKey + ":").getBytes(StandardCharsets.UTF_8));
+
+        return restClient.post()
+                .uri("v1/payments/confirm")
+                .header("Authorization", "Basic " + encodedKey)
```

Comment:
직접 하드코딩할 필요 없이 이미 정의된 상수를 활용해주면 좋을 것 같네요. 😉 

```suggestion
                .header(HttpHeaders.AUTHORIZATION, "Basic " + encodedKey)
```